File: BeeCrowd/ArvoreBinaria.java
import java.util.Scanner;

class No{
    int elemento;
    No esq, dir;

    public No(int elemento){
        this.elemento = elemento;
        this.esq = null;
        this.dir = null;
    }

    public No(int elemento, No esq, No dir){
        this.elemento = elemento;
        this.esq = esq;
        this.dir = dir;
    }
}

class Arvore{
    No raiz;

    public Arvore(){
        this.raiz = null;
    }

    public void inserir(int x){
        raiz = inserir(raiz, x);
    }

    private No inserir(No i, int x){
        if(i == null){
            i = new No(x);
        } else if(x < i.elemento){
            i.esq = inserir(i.esq, x);
        } else if(x > i.elemento){
            i.dir = inserir(i.dir, x);
        } else {
            throw new RuntimeException("Erro.");
        }

        return i;
    }

    public void prefixo(){
        System.out.print("Pre.: ");
        prefixo(raiz);
        System.out.println();
    }

    private void prefixo(No i){
        if(i != null){
            System.out.print(i.elemento + " ");
            prefixo(i.esq);
            prefixo(i.dir);
        }
    }

    public void infixo(){
        System.out.print("In..: ");
        infixo(raiz);
        System.out.println();
    }

    private void infixo(No i){
        if(i != null){
            infixo(i.esq);
            System.out.print(i.elemento + " ");
            infixo(i.dir);
        }
    }

    public void posfixo(){
        System.out.print("Post: ");
        posfixo(raiz);
        System.out.println();
    }

    private void posfixo(No i){
        if(i != null){
            posfixo(i.esq);
            posfixo(i.dir);
            System.out.print(i.elemento + " ");
        }
    }
}

public class ArvoreBinaria {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int testes = sc.nextInt();

        for (int i = 0; i < testes; i++) {
            Arvore arvore = new Arvore();
            int quantElementos = sc.nextInt();
            int elemento[] = new int[quantElementos];
            for (int j = 0; j < quantElementos; j++) {
                elemento[j] = sc.nextInt();
                arvore.inserir(elemento[j]);
            }

            System.out.println("Case " + (i + 1) + ":");
            arvore.prefixo();
            arvore.infixo();
            arvore.posfixo();
            
            System.out.println();
        }

        sc.close();
    }
}

File: BeeCrowd/ArvoreBinaria.c
#include <stdio.h>
#include <stdlib.h>

typedef struct No{
    int elemento;
    struct No *esq, *dir;
} No;

No* novoNo(int elemento){
    No* novo = (No*)malloc(sizeof(No));
    novo->elemento = elemento;
    novo->esq = NULL;
    novo->dir = NULL;
    return novo;
}

No* inserir(No* i, int x){
    if(i == NULL){
        i = novoNo(x);
    } else if(x < i->elemento){
        i->esq = inserir(i->esq, x);
    } else if(x > i->elemento){
        i->dir = inserir(i->dir, x);
    }
    return i;
}

void liberarArvore(No* raiz){
    if(raiz != NULL){
        liberarArvore(raiz->esq);
        liberarArvore(raiz->dir);
        free(raiz);
    }
}

int primeiro;

void prefixoRec(No* i){
    if(i != NULL){
        if (primeiro) {
            printf("%d", i->elemento);
            primeiro = 0;
        } else {
            printf(" %d", i->elemento);
        }
        prefixoRec(i->esq);
        prefixoRec(i->dir);
    }
}

void prefixo(No* raiz){
    prefixoRec(raiz);
}

void infixoRec(No* i){
    if(i != NULL){
        infixoRec(i->esq);
        if (primeiro) {
            printf("%d", i->elemento);
            primeiro = 0;
        } else {
            printf(" %d", i->elemento);
        }
        infixoRec(i->dir);
    }
}

void infixo(No* raiz){
    infixoRec(raiz);
}

void posfixoRec(No* i){
    if(i != NULL){
        posfixoRec(i->esq);
        posfixoRec(i->dir);
        if (primeiro) {
            printf("%d", i->elemento);
            primeiro = 0;
        } else {
            printf(" %d", i->elemento);
        }
    }
}

void posfixo(No* raiz){
    posfixoRec(raiz);
}

int main(){
    int testes;
    scanf("%d", &testes);

    for(int i = 0; i < testes; i++){
        No* raiz = NULL;
        int quantElementos;
        scanf("%d", &quantElementos);
        int elemento[quantElementos];

        for(int j = 0; j < quantElementos; j++){
            scanf("%d ", &elemento[j]);
            raiz = inserir(raiz, elemento[j]);
        }

        printf("Case %d:\n", i + 1);

        primeiro = 1;
        printf("Pre.: ");
        prefixo(raiz);
        printf("\n");

        primeiro = 1;
        printf("In..: ");
        infixo(raiz);
        printf("\n");

        primeiro = 1;
        printf("Post: ");
        posfixo(raiz);
        printf("\n\n");

        liberarArvore(raiz);
    }

    return 0;
}
